Mage/src/main/java/mage/abilities/keyword/MutateAbility.java

```diff
@@ -1,39 +1,191 @@
+    protected static final String MUTATE_REMINDER = "<i>(If you cast this spell for its mutate cost, "
+            + "put it over or under target non-Human creature you own. "
+            + "They mutate into the creature on top plus all abilities from under it.)</i>";
+    protected Costs<Cost> mutateCosts;
+    protected String ruleText;
+
+    public MutateAbility(Card card, String mutateCost) {
+        this(card, createCosts(new ManaCostsImpl<>(mutateCost)));
+    }
+
+    public MutateAbility(Card card, Costs<Cost> mutateCosts) {
+        super(Zone.ALL, null);
+        this.mutateCosts = mutateCosts;
+        this.name = MUTATE_KEYWORD;
+        this.activated = false;
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(MUTATE_KEYWORD).append(' ');
+        for (Cost cost : mutateCosts) {
+            if (!(cost instanceof ManaCosts)) {
+                sb.setLength(sb.length() - 1);
+                sb.append("&mdash;");
+                break;
+            }
+        }
+        sb.append(mutateCosts.getText());
+        if (!(mutateCosts.get(mutateCosts.size() - 1) instanceof ManaCosts)) {
+            sb.append('.');
+        }
+        sb.append(' ').append(MUTATE_REMINDER);
+
+        this.ruleText = sb.toString();
     }
 
     private MutateAbility(final MutateAbility ability) {
         super(ability);
+        this.ruleText = ability.ruleText;
+        this.mutateCosts = ability.mutateCosts;
+    }
+
+    private static Costs<Cost> createCosts(Cost cost) {
+        Costs<Cost> costs = new CostsImpl<>();
+        costs.add(cost);
+        return costs;
     }
 
     @Override
     public MutateAbility copy() {
         return new MutateAbility(this);
     }
 
+    @Override
+    public boolean isActivated(Ability source, Game game) {
+        return activated;
+    }
+
+    @Override
+    public String getCastMessageSuffix(Game game) {
+        return " with " + mutateCosts.getText();
+    }
+
+    @Override
+    public boolean isAvailable(Ability source, Game game) {
+        return source.getZone().equals(Zone.HAND) || source.getZone().equals(Zone.COMMAND);
+    }
+
+    @Override
+    public boolean askToActivateAlternativeCosts(Ability ability, Game game) {
+        if (ability.getAbilityType() == AbilityType.SPELL) {
+            Player player = game.getPlayer(ability.getControllerId());
+            Spell spell = game.getStack().getSpell(ability.getId());
+            if (player != null && spell != null) {
+                if (mutateCosts.canPay(ability, this, ability.getControllerId(), game)) {
+                    if (player.chooseUse(Outcome.Benefit,
+                            "Cast " + spell.getLogName() + " with mutate for " + getCosts().getText() + " ?", ability, game)) {
+                        mutateCosts.clearPaid();
+                        TargetCreaturePermanent target = new TargetCreaturePermanent(filter);
+                        target.setNotTarget(true);
```

Comment:
`setNotTarget` is wrong. Creature spells cast for their mutate cost are targetted spells in every sense, just like Auras. For example, you can't mutate onto a creature with shroud or with protection from one of the mutate card's characteristics.

Reply:
Woops, good catch!

Reply:
Fixed by reworking `MutateAbility` to extend `SpellAbility` for all of the casting logic.